crates/db_schema/src/impls/combined/modlog.rs

```diff
@@ -0,0 +1,46 @@
+use crate::{
+  diesel::{ExpressionMethods, QueryDsl, SelectableHelper},
+  newtypes::PaginationCursor,
+  schema::modlog_combined,
+  source::combined::modlog::ModlogCombined,
+  traits::PageCursorReader,
+  utils::DbConn,
+};
+use diesel_async::RunQueryDsl;
+use lemmy_utils::error::{LemmyErrorType, LemmyResult};
+
+#[async_trait]
+impl PageCursorReader for ModlogCombined {
+  async fn from_cursor(cursor: PaginationCursor, conn: &mut DbConn<'_>) -> LemmyResult<Self> {
+    let (prefix, id) = cursor.prefix_and_id()?;
+
+    let mut query = modlog_combined::table
+      .select(Self::as_select())
+      .into_boxed();
+
+    query = match prefix {
+      'A' => query.filter(modlog_combined::admin_allow_instance_id.eq(id)),
+      'B' => query.filter(modlog_combined::admin_block_instance_id.eq(id)),
+      'C' => query.filter(modlog_combined::admin_purge_comment_id.eq(id)),
```

Comment:
For the modlog, which has like 17 tables, I changed these from strings to chars.